File: digital-wallet-frontend/src/services/authService.ts
import api from '../api';
import { RegisterRequest, LoginRequest } from '../types/auth';

export const authService = {
    login: async (data: LoginRequest) => {
        const response = await api.post('/auth/login', data);
        return response.data;
    },
    register: async (data: RegisterRequest) => {
        const response = await api.post('/auth/register', data);
        return response.data;
    },
    refresh: async (refreshToken: string) => {
        const response = await api.post('/auth/refresh', { refreshToken });
        return response.data;
    },
    logout: async () => {
        await api.post('/auth/logout');
    },
};

File: digital-wallet-backend/src/main/java/com/alimertkaya/digitalwallet/auth/service/AuthService.java
package com.alimertkaya.digitalwallet.auth.service;

import com.alimertkaya.digitalwallet.auth.dto.AuthResponse;
import com.alimertkaya.digitalwallet.auth.dto.LoginRequest;
import com.alimertkaya.digitalwallet.auth.dto.RegisterRequest;
import reactor.core.publisher.Mono;

public interface AuthService {

    // yeni kullaniciyi kaydetme
    // @param request kayit bilgileri (username, password. email etc.) iceren DTO
    // return JWT token iceren AuthResponse
    Mono<AuthResponse> register(RegisterRequest request);

    // @param request giris bilgileri iceren DTO
    // @return JWT token iceren AuthResponse
    Mono<AuthResponse> login(LoginRequest request);

    Mono<AuthResponse> refreshToken(String refreshToken);
}